dove/src/cmd/prover.rs

```diff
@@ -83,18 +138,57 @@ fn is_z3_available(z3_exe: &str) -> bool {
     is_executable_available(z3_exe, &["-h"])
 }
 
+fn is_cvc4_available(cvc4_exe: &str) -> bool {
+    is_executable_available(cvc4_exe, &["-V"])
+}
+
 /// Checks if executable is available in path by running it.
 fn is_executable_available<S: AsRef<OsStr>, I: IntoIterator<Item = S>>(
     executable: &str,
     args: I,
 ) -> bool {
-    let status = std::process::Command::new(executable)
-        .args(args)
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
-        .status();
-    match status {
-        Ok(status) => status.success(),
+    let result = std::process::Command::new(executable).args(args).output();
+
+    match result {
+        Ok(result) => {
+            let status_success = result.status.success();
+            if !status_success {
+                println!("Warning: {}", String::from_utf8(result.stderr).unwrap());
+            }
+            result.status.success()
+        }
         Err(_) => false,
     }
 }
+
+/// get path to z3
+fn get_path_z3() -> Result<String, anyhow::Error> {
+    let env_path = env!("PATH");
+
+    #[cfg(not(target_family = "windows"))]
+    let separator = ':';
+
+    #[cfg(target_family = "windows")]
+    let separator = ':';
+
+    env_path
+        .split(separator)
+        .map(PathBuf::from)
+        .map(|path| path.join(Z3_EXE))
+        .find(|path|path.exists())
+        .map_or(
+            Err(anyhow!("z3 executable not found in PATH. Please install it from https://github.com/Z3Prover/z3")),
+                |path| path.to_str().ok_or_else(||anyhow!("couldn't convert path to string")).map(|t|t.to_string())
```

Comment:
Rustfmt is too forgiving on this line...

Reply:
@vitvakatu Thanks. Fixed